### ISSUE TITLE
XmlParser Bug

### ISSUE THREAD

**Issue by kids-return** [yiisoft/yii2-httpclient#102]
### What steps will reproduce the problem?
```php
        $content = '<xml>
                    <head>
                        <title>parseXml</title>
                    </head>
                    <body>
                        <lists name="attribute">
                            <list>
                                <id>1</id>
                                <name>name1</name>
                            </list>
                            <list>
                                <id>1</id>
                                <name>name2</name>
                            </list>
                        </lists>
                    </body>
                </xml>';
        $array = $this->convertXmlToArray($content);
        var_dump($array);
        exit();
```
### What's expected?

```php
array(2) {
  ["head"]=>
  array(1) {
    ["title"]=>
    string(8) "parseXml"
  }
  ["body"]=>
  array(1) {
    ["lists"]=>
    array(2) {
      ["@attributes"]=>
      array(1) {
        ["name"]=>
        string(9) "attribute"
      }
      ["list"]=>
      array(2) {
        [0]=>
        array(2) {
          ["id"]=>
          string(1) "1"
          ["name"]=>
          string(5) "name1"
        }
        [1]=>
        array(2) {
          ["id"]=>
          string(1) "1"
          ["name"]=>
          string(5) "name2"
        }
      }
    }
  }
}
```

### What do you get instead?
Object
```php
array(2) {
  ["head"]=>
  array(1) {
    ["title"]=>
    string(8) "parseXml"
  }
  ["body"]=>
  array(1) {
    ["lists"]=>
    array(2) {
      ["@attributes"]=>
      array(1) {
        ["name"]=>
        string(9) "attribute"
      }
      ["list"]=>
      array(2) {
        [0]=>
        object(SimpleXMLElement)#2530 (2) {  <---------------object
          ["id"]=>
          string(1) "1"
          ["name"]=>
          string(5) "name1"
        }
        [1]=>
        object(SimpleXMLElement)#2531 (2) {
          ["id"]=>
          string(1) "1"
          ["name"]=>
          string(5) "name2"
        }
      }
    }
  }
}
```

### Additional info

| Q                       | A
| ----------------------- | ---
| Yii version             | 2.0.12
| Yii HTTP Client version | 2.0.3
| PHP version             | 5.6
| Operating system        | windows

**Comment by kids-return** [comment]
solution :
```php
    protected function convertXmlToArray($xml)
    {
        if (is_string($xml)) {
            $xml = simplexml_load_string($xml, 'SimpleXMLElement', LIBXML_NOCDATA);
        }
        $result = (array) $xml;
        foreach ($result as $key => $value) {
            if (is_object($value) || is_array($value)) {
                $result[$key] = $this->convertXmlToArray($value);
            }
        }
        return $result;
    }
```

**Comment by klimov-paul** [comment]
Resolved by commit 44c429a2f67b036a8c3654886420fcdf1252779a
